### DISCUSSION TITLE
Trouble with parsing basic markdown in react

### DISCUSSION THREAD

**Discussion by fegigy1** [markdoc/markdoc#92] [Q&A] [answered]
Dear Team,

I am ` beginner to the markdown world. I am trying to use markdoc in my react application (with typescript) . I am facing a basic issue of parsing simple markdown content. I am following below example to setup but my application uses with React 17 for some specific reasons.

https://markdoc.io/docs/examples/react

Below is my markdown file :

```
	# Test H1
	## Test H2
```

I am using below code to parse markdown file but I am not getting the content with expected tags. Could someone help me if I am missing anything here?

Server side :

```
  const document = fs.readFileSync(CONTENT_DIR + "/test.md", "utf-8");
  const ast = Markdoc.parse(document);
  content = Markdoc.transform(ast, {});
  return res.json(content);
```

Client side :

```
	const content = await response.json();
	return Markdoc.renderers.react(content, React, {}) as any;
```

Note: When I see AST all the nodes have tags undefined.

Any help would be appreciated.

Thanks.

**Comment by mfix-stripe** [comment, marked as answer]
Hey @fegigy1, you ran into the issue described here: https://github.com/markdoc/markdoc/issues/82, sorry about that.
We are currently looking into the fix and hope to have it up shortly. 

You can fix the issue above by removing the indentation:

```md
# Test H1
## Test H2
```

**Reply by fegigy1** [reply]
Thank you for the reply @mfix-stripe. But unfortunately removing indentation is also not working.

**Reply by mfix-stripe** [reply]
@fegigy1 could you share a reproduction of it not working without indentation? I'm certain regular headings work:

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/62121649/174676757-e425c696-6b89-4619-97ce-0d639d0598a4.png">

**Reply by fegigy1** [reply]
Sure @mfix-stripe . I have observed that the output is rendered correct with React 18 but not with React 17. PFB my configurations and outputs if it helps to figure out something. TIA.

![image](https://user-images.githubusercontent.com/107616072/174754363-37fdf93b-36fd-46d3-a84d-db78d7c0f0b2.png)

![image](https://user-images.githubusercontent.com/107616072/174754470-2a5b1280-3e13-4ffc-8721-cc8a04acbd0f.png)

**Reply by mfix-stripe** [reply]
Ahh yes, @fegigy1, the example above is based on React 18, as you can see here: https://github.com/markdoc/docs/blob/main/examples/react-nodejs/package.json#L10-L11

If you want to use React 17, I think you just need to adjust a couple create-react-app things, but that is entirely separate from Markdoc. Markdoc does not depend on any specific React version.
